web/components/visibility-observer.tsx

```diff
@@ -8,17 +8,18 @@ export function VisibilityObserver(props: {
   const { className } = props
   const [elem, setElem] = useState<HTMLElement | null>(null)
   const onVisibilityUpdated = useEvent(props.onVisibilityUpdated)
-
-  useEffect(() => {
-    const hasIOSupport = !!window.IntersectionObserver
-    if (!hasIOSupport || !elem) return
-
-    const observer = new IntersectionObserver(([entry]) => {
+  const observer = useRef(
+    new IntersectionObserver(([entry]) => {
```

Comment:
This actually creates a new object on every render, and all but the first are discarded. Doesn't seem that much better. You could do the use state with init function to solve this. Otherwise, I don't think it really matters.

Reply:
Huh, TIL. Well, I won't bother changing it to something uglier unless it ever becomes an actual problem.